### PR TITLE
Multipage selection

### LINKED ISSUE #168: Multipage Selection
This is related to the subsequent conversation in #69.

Thank you for implementing automatic PDF import!  I was just testing it out, and I see that it only accesses page 1 (I believe at this line of code: https://github.com/ankitrohatgi/WebPlotDigitizer/blob/c229d056597d95ff7d7ac21555765ca0e26986c6/app/javascript/controllers/imageManager.js#L61).  For multipage image formats (.pdf and .tiff immediately come to my mind, and there are probably others), it would be handy to have the ability to switch between pages.

I think that this could be implemented under the "Image" info section at the top left where instead of just giving pixels it could give pixels and pages, and there could be a page selector in there, too.

I'm not sure how efficient PDFJS is at managing files, but I occasionally work with .pdf files that are multiple thousand pages long, and it'd be handy to be able to use those directly (maybe just hold the raw file in memory and use PDFJS only on a specific page at a time?).

On the JSON side, everywhere there is now an x and y coordinate, it would also need a page number, but hopefully that's it.

### PR DESCRIPTION
**Pull request by ewchow**
Took a swing at adding support for multi-page PDFs (resolves #168). I kept the PDF management logic somewhat generalized, in case support for other file types are necessary in the future (TIFFs perhaps). Also tried to keep the existing logic as untouched as I could.

Main changes in this PR:
- Added UI page navigation elements
  - Added _Page X of Y_ display under _Image_ tree
    image
    - Clicking on _Page X of Y_ redirects selection to _Image_
    - Only applicable to multi-page PDFs
  - Added _Previous Page_, _Next Page_, and _Go to page: X_ controls in the _Image_ tree widget
    image
- Added multi-page support to axes, datasets, and measurements
  - Output JSON structure includes additional _page_ key for all 3 sets
  - Only sets associated with current page are displayed in the UI
- Added multi-page support for undoing/redoing image cropping
- Added multi-page support for saving/loading JSON files
- Added PDF support for saving/loading project files 

I also took the liberty to add a `.gitignore` file. Happy to make any changes you'd like. Let me know what you think.

Happy Holidays!

### CODE REVIEW

File: app/javascript/core/axes/bar.js [hunk 1]
```diff
@@ -154,6 +154,10 @@ wpd.BarAxes = (function() {
         };
 
         this.name = "Bar";
+
+        if (wpd.appData.isMultipage()) {
```
Comment (ankitrohatgi) [inline app/javascript/core/axes/bar.js]: I'd like to keep `core` pieces independent of the higher level `appData` as much as possible. Can we keep `page` out of both `axes` and `dataset`?

**Comment by ankitrohatgi** [comment]
Hey, thanks again for the contributions and happy holidays to you too!

This looks good for most part - just the `axes.page`, `ds.page` and `ms.page` seem like maybe they need another thought. I have just superficially scanned though for now - I will take a closer look in the next few days.

File: app/javascript/core/axes/bar.js (hunk 1 above)
Comment (ewchow) [reply app/javascript/core/axes/bar.js]: Sure, I'll work on moving them out

**Comment by ewchow** [comment]
Hey Ankit, hope you're well!

Finally got around to finishing up the refactor. I've moved the calls to `appData` out of `core` and removed `axes.page`, `ds.page`, and `ms.page` from the object definitions. The page managing logic is now more centralized in `pageManager`. The saved JSON will programmatically include the page info if it exists, otherwise it'll be as it were before.

**Comment by ankitrohatgi** [comment]
Hi @ewchow, thanks for making the changes I requested. The new version looks good!

I see you have a few conflicts to fix - if you can resolve those, we can merge this :)

**Comment by ewchow** [comment]
Great! The conflicts are resolved now. All set, thanks!

**Pull request merged**